python/langsmith/_testing.py

```diff
@@ -316,15 +317,28 @@ def _start_experiment(
         return client.read_project(project_name=experiment_name)
 
 
-def _get_id(func: Callable, inputs: dict, suite_id: uuid.UUID) -> uuid.UUID:
+# Track the number of times a parameter has been used in a test
+# This is to ensure that we can uniquely identify each test case
+# defined using pytest.mark.parametrize
+_param_dict: dict = defaultdict(lambda: defaultdict(int))
```

Comment:
Global state not great, and being order based means if you insert an example inline the other examples will shift down to replace the others' ids, but:

1. Naming is cleaner and doens't require the parameters to be serializable.
2. If the example is updated, we get different versions so presumably this handles the "apples to apples" comparison issues